bitrise.yml

```diff
@@ -146,6 +148,30 @@ workflows:
         - scheme: Fennec_Enterprise_XCUITests
         - xcodebuild_test_options: "-testPlan Fennec_Enterprise_XCUITests"
         - simulator_device: "$IOS_DEVICE"
+  Detect_warnings:
+    steps:
+    - script@1.1:
+        title: Detect number of warnings
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            # fail if any commands fails
+            set -e
+            # debug log
+            set -x
+
+            TEST_LOG_PATH=$BITRISE_XCODEBUILD_TEST_LOG_PATH
+
+            COUNT_XCUI=$(./test-fixtures/generate-metrics.sh $TEST_LOG_PATH all)
+            envman add --key SHOW_WARNING_COUNT_XCUI --value "$COUNT_XCUI"
+
+            STR=$COUNT_XCUI
+            SUB='greater'
+            if [[ "$STR" == *"$SUB"* ]]; then
```

Comment:
The way to fail the step is to exit the script with error, and I have thought about checking the output of the generate-metrics.sh to decide if the number is lower or higher than threshold